### ISSUE TITLE
Consider wrapping throw in call to encourage JITter

### ISSUE THREAD

**Issue by mtmk** [nats-io/nats.net#320]
NIT/Future: Consider wrapping throw in call to encourage JITter.

_Originally posted by @to11mtm in https://github.com/nats-io/nats.net.v2/pull/303#discussion_r1442269273_

**Comment by to11mtm** [comment]
So, to elaborate here... Apologies, priorities abound...

In general we should, when throwing in potentially hot methods, prefer to call a `ThrowHelper()` instead of throwing.

As a very good example https://github.com/nats-io/nats.net.v2/blob/421463c4c9bec2bd4c88b5f3ee70fd11c84d3422/src/NATS.Client.Core/Internal/NatsReadProtocolProcessor.cs#L76-L85;

the fact we are throwing actually causes a number of optimizations to not happen and can also impact inline ability.

An example refactoring would be:
```
    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    private static int GetInt32(ReadOnlySpan<byte> span)
    {
        if (!Utf8Parser.TryParse(span, out int value, out var consumed))
        {
            ThrowOnUtf8ParseFail();
        }

        return value;
    }
    [MethodImpl(MethodImplOptions.NoInlining)]
    private static void ThrowOnUtf8ParseFail()
    {
        throw new Exception();
    }
```

**Comment by to11mtm** [comment]
Further Docs and explanation of potential value for perf: (FWIW these are from CommunityToolkit's Diagnostic analyzer docs, so we could, if we wanted, add as an analyzer to help identify across projects)

https://learn.microsoft.com/en-us/dotnet/communitytoolkit/diagnostics/throwhelper#technical-details

**Comment by mtmk** [comment]
Makes sense. Also runtime libs are always using this. We could start with read/write loops maybe?
